tests/api/printf.cpp

```diff
@@ -104,6 +106,36 @@ static std::string getStdoutFileName(temp_folder_deletion& temp) {
     return (prefix / suffix).string();
 }
 
+void printf_callback(const char* buffer, size_t len, size_t complete,
```

Comment:
This function is too complicated I feel.
Let's make `globalBuffer` a `std::string` it should make it much simplier.